lib/plugins/threaded-signing/plugin.c

```diff
@@ -172,13 +216,8 @@ signing_worker_thread(void *user_data)
 /* This function is called from the library upon signing and the input |signature_info| includes
  * all necessary information to do so.
  *
- * The |hash| is copied to |hash_to_sign| and the |plugin_state| is moved to
- * THREADED_SIGNING_HAS_HASH_TO_SIGN. If this is the first time of signing, memory for
- * |self->signature_info| and |hash_to_sign| is allocated and the |private_key| is copied from
- * |signature_info|.
- *
- * Further, if there is no worker thread running or if the latest signing has not yet finished an
- * SV_NOT_SUPPORTED is returned. */
+ * The |hash| is copied to |input_buffer|. If this is the first time of signing, memory for
+ * |input_buffer| is allocated and the |private_key| is copied from |signature_info|. */
```

Comment:
I guess you still mean ```self->signature_info``` and not ```input_buffer```. The ```input_buffer``` itself was allocated when ```self``` was created. What should be added to the description though is something like "If memory for the hash has not been allocated it will be allocated."
You also need to describe what happens if we run out of hash slots in ```input_buffer```.